pkg/controllers/dynakube/extension/service.go

```diff
@@ -0,0 +1,96 @@
+package extension
+
+import (
+	"context"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/labels"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/service"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (r *reconciler) reconcileService(ctx context.Context) error {
+	if r.dk.PrometheusEnabled() {
+		log.Info("reconcile service")
+
+		if err := r.ensureService(ctx); err != nil {
+			return errors.WithMessage(err, "could not update service")
+		}
+
+		return nil
+	} else {
+		svc, err := r.buildService()
+		if err != nil {
+			log.Error(err, "could not build service during cleanup")
+		}
+
+		err = service.Query(r.client, r.apiReader, log).Delete(ctx, svc)
+
+		if err != nil {
+			log.Error(err, "failed to clean up extension service")
+
+			return nil
+		}
+	}
+
+	return nil
+}
+
+func (r *reconciler) ensureService(ctx context.Context) error {
+	_, err := service.Query(r.client, r.apiReader, log).Get(ctx, client.ObjectKey{Name: r.buildServiceName(), Namespace: r.dk.Namespace})
+
+	if k8serrors.IsNotFound(err) {
+		log.Info("service was not found, creating service")
+
+		return r.createService(ctx)
+	} else if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
unnecessary, the `query.CreateOrUpdate` already handles this scenario ([code](https://github.com/Dynatrace/dynatrace-operator/blob/4a5d1a041c3f20460973ba8dd662653b982f957c/pkg/util/kubeobjects/internal/query/query.go#L71-L83))

Reply:
fixed in 8adba07f